app/views/admin/users/_add_membership_button_if_needed.html.arb

```diff
@@ -0,0 +1,17 @@
+if (entity_type == 'entry' &&
+    Pageflow::EntryPolicy::Scope.new(user, Pageflow::Entry).member_addable &&
+    !membership_entries_collection(resource,
+                                   Pageflow::Membership.new(user: resource),
+                                   Pageflow::Membership.new).blank?) ||
+   (entity_type == 'account' &&
+    Pageflow::AccountPolicy::Scope.new(user, Pageflow::Account).member_addable &&
+    !membership_accounts_collection(resource,
+                                    Pageflow::Membership.new(user: resource),
+                                    Pageflow::Membership.new).blank?)
```

Comment:
This huge condition should live inside a helper  where we can extract methods. Maybe the whole partial could instead be a Arbre view component.